packages/grafana-ui/src/components/Table/TableNG/components/SummaryCell.tsx

```diff
@@ -72,43 +74,34 @@ export const SummaryCell = ({
   );
   const SummaryCellValue = ({ children }: { children: ReactNode }) => (
     <div
-      data-testid={selectors.components.Panels.Visualization.TableNG.Footer.Value}
+      data-testid={selectors.components.Panels.Visualization.TableNG.Footer[children == null ? 'EmptyValue' : 'Value']}
       className={styles.footerItemValue}
     >
-      {children}
+      {children ?? <>&nbsp;</>}
     </div>
   );
 
-  // Render each reducer in the footer
+  const defaultFooterCellStyles = getDefaultCellStyles(theme, {
+    textAlign: 'left', // alignment is set in footerItem
+    shouldOverflow: true,
+    textWrap: false,
+  });
+
   return (
     <div
-      className={cellClass}
-      data-testid={reducerResultsEntries.length === 0 && !renderRowLabel ? 'summary-cell-empty' : undefined}
+      className={clsx(styles.footerCell, defaultFooterCellStyles)}
+      data-testid={entries.length === 0 ? 'summary-cell-empty' : undefined}
     >
-      {reducerResultsEntries.map(([reducerId, reducerResult]) => {
-        // empty reducer entry, but there may be more after - render a spacer.
-        if (reducerResult === null) {
-          return (
-            <SummaryCellItem key={reducerId}>
-              {rowLabel ? <SummaryCellLabel>{getReducerName(reducerId)}</SummaryCellLabel> : <>&nbsp;</>}
-            </SummaryCellItem>
-          );
-        }
-
+      {entries.map(([reducerId, reducerResult], idx) => {
         return (
-          <SummaryCellItem key={reducerId}>
-            {!hideLabel && <SummaryCellLabel>{getReducerName(reducerId)}</SummaryCellLabel>}
+          <SummaryCellItem key={reducerId} idx={idx}>
+            {(reducerResult != null || rowLabel) && !hideLabel && (
+              <SummaryCellLabel>{getReducerName(reducerId)}</SummaryCellLabel>
+            )}
```

Comment:
this looks like more changes than it is... ultimately, I was able to DRY up this code by consolidating down to a single `entries` array in a memo which handles the default case as well as the `rowLabel` case when there are no reducers in this cell.